creator-node/src/routes/tracks.js

```diff
@@ -50,29 +56,122 @@ module.exports = function (app) {
     syncLockMiddleware,
     handleTrackContentUpload,
     handleResponse(async (req, res) => {
-      const AsyncProcessingQueue =
-        req.app.get('serviceRegistry').asyncProcessingQueue
-
       if (req.fileSizeError || req.fileFilterError) {
         removeTrackFolder({ logContext: req.logContext }, req.fileDir)
         return errorResponseBadRequest(req.fileSizeError || req.fileFilterError)
       }
 
-      await AsyncProcessingQueue.addTrackContentUploadTask({
+      const AsyncProcessingQueue =
+        req.app.get('serviceRegistry').asyncProcessingQueue
+
+      const currentNodeCanHandleTranscode = canCurrentNodeHandleTranscode({
+        transcodingQueueCanAcceptMoreJobs: await TranscodingQueue.isAvailable(),
+        spID: config.get('spID'),
+        libs: AsyncProcessingQueue.getLibs()
```

Comment:
is the only reason we're attaching libs/a getter/setter interface for it to AsyncProcessingQueue for compatibility here?

Reply:
oh actually, this check is actually kind of misleading. should be refactored to see if the current node should handle the transcode or not. changed in PR